src/service/filtering/filtering_processing.py

```diff
@@ -0,0 +1,168 @@
+            field,
+            column.type,
+            querystring,
+            analyzers.get_analyzer(column.filter_strategy),
+            column.filter_strategy
+        )
+    return fs
+
+
+async def get_filters(
+        r: Request,
+        arango_coll: str,
+        coll_id: str,
+        load_ver: str,
+        load_ver_override: bool,
+        data_product: str,
+        columns: list[col_models.AttributesColumn],
+        view_name: str = None,
+        count: bool = False,
+        sort_on: str = None,
+        sort_desc: bool = False,
+        filter_conjunction: bool = True,
+        match_spec: SubsetSpecification = None,
+        selection_spec: SubsetSpecification = None,
+        keep: dict[str, set[col_models.ColumnType]] = None,
+        keep_filter_nulls: bool = False,
+        skip: int = 0,
+        limit: int = 1000,
+        start_after: str = None,
+        trans_field_func: Callable[[str], str] = None,
+) -> FilterSet:
+    """
+    Constructs a FilterSet and applies filters using the provided filter query and columns.
+
+    Requires a 'get_columns_func' to generate columns, represented as a dictionary mapping column name/ID to
+    columnar_attribs_common_models.AttributesColumn objects.
+
+    r - The request.
+    arango_coll - The name of the Arango collection.
+    coll_id - The collection ID.
+    load_ver - The load version.
+    load_ver_override - Whether or not to override the load version.
+    data_product - The data product id.
+    columns - list of columnar_attribs_common_models.AttributesColumn objects to be filtered on.
+    view_name - The name of the view to use for filtering.
+    count - Whether or not to return the count of matching documents.
+    sort_on - The name of the field to sort on.
+    sort_desc - Whether or not to sort in descending order.
+    filter_conjunction - Whether or not to use a conjunction for filtering.
+    match_spec - A subset specification for matching.
+    selection_spec - A subset specification for selection.
+    keep - A dictionary mapping column names to sets of column types to keep.
+    keep_filter_nulls - Whether or not to keep null values when filtering.
+    skip - The number of documents to skip.
+    limit - The maximum number of documents to return.
+    start_after - The value of the field to start after.
+    trans_field_func - A function to transform the field name to valid column name in the filter query
+    """
+    appstate = app_state.get_app_state(r)
+    filter_query = _get_filter_map(r)
+    if filter_query:
+        if load_ver_override:
+            # If we need this feature than the admin needs to supply the view name to use
+            # via the API
+            raise ValueError("Filtering is not supported with a load version override.")
+
+        if view_name:
+            if not await appstate.arangostorage.has_search_view(view_name):
+                raise ValueError(f"View {view_name} does not exist for collection {coll_id}")
```

Comment:
```suggestion
                raise ValueError(f"View {view_name} does not exist for data product {data_product}")
```

Views aren't KBase Collection specific. We might want to put in the arango collection... not sure if that's a help or not seeing as currently there's a 1:1 relationship from dp <-> arango collection w/ view